src/v/storage/tests/storage_e2e_test.cc

```diff
@@ -4555,6 +4555,26 @@ FIXTURE_TEST(test_offset_range_size2_compacted, storage_test_fixture) {
       == std::nullopt);
 };
 
+namespace storage {
+class segment_index_observer {
+public:
+    explicit segment_index_observer(segment_index& index)
+      : _index{index} {}
+
+    size_t max_step_size() const {
+        fragmented_vector<size_t> diffs;
+        auto& pos = _index._state.position_index;
+        diffs.reserve(pos.size());
+        std::adjacent_difference(
+          pos.begin(), pos.end(), std::back_inserter(diffs));
```

Comment:
didn't know about `adjacent_difference`, cool

Reply:
Unfortunately it does not work on const references to containers, but handy where usable.